distributed_shampoo/fsdp_shampoo.py

```diff
@@ -353,6 +361,7 @@ def _initialize_preconditioners_and_steps(self):
         # current workaround for group_source_rank
         # TODO: try to keep dist_group information
         self._dist_group = None
+        group_rank = dist.get_rank()
```

Comment:
This is probably a remnant from when I was using BlockShampooPreconditioner. I am not currently using these variables in a meaningful way, but it may be a good idea to pass them into the communication preconditioner class in the future (to enable HSDP, as you said).

Reply:
Just to be clear, it's likely the case if we are using HSDP that we need to specify a process group when instantiating `FSDPShampoo`. In that case, we probably need to change this to `group_rank = dist.get_rank(process_group)`, which will then inform the assignment below.